### PR TITLE
udpbroadcastrelay - UDP unicast/multicast relayer  for the likes of Sonos etc. 

### PR DESCRIPTION
**Pull request by marjohn56**
udpbroadcastrelay. A daemon to handle forwarding of udp unicast and multicast packets between interfaces for use with Sonos, Sky, Chromecast and others.

This version varies from the original UDP-BROADCAST-RELAY-REDUX in that it now generates its own pid file when forking to the background, and should also clean up and delete the pid file after it exits, it does in all the tests I have carried out.

### CODE REVIEW

File: net/udpbroadcastrelay/Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+udpbroadcastrelay: main.c
+	gcc -g main.c -o udpbroadcastrelay
```
Comment (fabianfrz) [inline net/udpbroadcastrelay/Makefile]: should be the variable CC which can be gcc or clang in most cases - in linux this is GCC, on BSD it is LLVM (clang).

File: net/udpbroadcastrelay/main.c [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+    0,    0,    0,    0,
+    0,    0,    0,    0,
+    0x00,    0x12,    0x00,    0x00,
+    '1','2','3','4','5','6','7','8','9','0'
+};
+
+void inet_ntoa2(struct in_addr in, char* chr, int len) {
+    char* from = inet_ntoa(in);
+    strncpy(chr, from, len);
+}
```
Comment (fabianfrz) [inline net/udpbroadcastrelay/main.c]: I think this is short enough to allow inlining.

File: net/udpbroadcastrelay/main.c [hunk 3]
```diff
@@ -0,0 +1,651 @@
...
+
+void catch_sigterm()
+{
+    static struct sigaction _sigact;
+
+    memset(&_sigact, 0, sizeof(_sigact));
+    _sigact.sa_sigaction = sig_term_handler;
+    _sigact.sa_flags = SA_SIGINFO;
+
+    sigaction(SIGTERM, &_sigact, NULL);
```
Comment (fabianfrz) [inline net/udpbroadcastrelay/main.c]: can you remove the underscore? It looks awful.

File: net/udpbroadcastrelay/main.c [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+        );
+
+        // Set up a one raw socket per interface for sending our packets through
+        if((iface->raw_socket = socket(AF_INET,SOCK_RAW,IPPROTO_RAW)) < 0) {
+            perror("socket");
+            exit(1);
+        }
+        {
+            int yes = 1;
+            int no = 0;
```
Comment (fabianfrz) [inline net/udpbroadcastrelay/main.c]: C99 defines booleans in stdbool.h as 0 and 1, so you can use true and false if you want.

http://openbook.rheinwerk-verlag.de/c_von_a_bis_z/005_c_basisdatentypen_015.htm

File: net/udpbroadcastrelay/main.c [hunk 5]
```diff
@@ -0,0 +1,651 @@
...
+                0,
+                (struct sockaddr*)&sendAddr,
+                sizeof(sendAddr)
+            ) < 0) {
+                perror("sendto");
+            }
+        }
+        DPRINT ("\n");
+    }
+}
```
Comment (fabianfrz) [inline net/udpbroadcastrelay/main.c]: the main function is far too big to be readable and understandable. It should be split up in smaller functions if possible.

File: net/udpbroadcastrelay/Makefile (hunk 1 above)
Comment (marjohn56) [reply net/udpbroadcastrelay/Makefile]: Is done.

File: net/udpbroadcastrelay/main.c (hunk 2 above)
Comment (marjohn56) [reply net/udpbroadcastrelay/main.c]: Not sure what you mean.

File: net/udpbroadcastrelay/main.c (hunk 3 above)
Comment (marjohn56) [reply net/udpbroadcastrelay/main.c]: Which one, there are several. I got this piece of code from a reference site. :)

File: net/udpbroadcastrelay/main.c (hunk 4 above)
Comment (marjohn56) [reply net/udpbroadcastrelay/main.c]: Done

File: net/udpbroadcastrelay/main.c (hunk 5 above)
Comment (marjohn56) [reply net/udpbroadcastrelay/main.c]: I'll think about that one, not my code to start with. :)

I've spent a lot more time on the plugin than this.

**Comment by mimugmail** [comment]
I think we can close this one here ...

**Pull request closed without merging**